lib/inc/internal/util/aix/odm.hpp

```diff
@@ -0,0 +1,263 @@
+#include <odmi.h>
+
+#include <cstdint>
+#include <memory>
+#include <string>
+
+namespace facter { namespace util { namespace aix {
+
+    // This is a singleton class that handles initialization and
+    // termination of the ODM library. As long as any other class holds a
+    // reference to this one, ODM will remain initialized.
+    //
+    // This class also provides static helpers for interacting with the ODM
+    class odm {
+    public:
+         typedef std::shared_ptr<odm> ptr;
+         static ptr get() {
+             static std::weak_ptr<odm> self;
```

Comment:
This is basically just implementing the lazy-singleton pattern, but on top of a weak ptr so it deallocates when it loses all references (instead of a true static member variable of the type we return). It seemed like the best way to handle "I can only have one of this thing, but I still want RAII-like initialization and cleanup"